// @coreui/react/es/toast/CToastHeader.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import CButtonClose from '../button/CButtonClose';
import { Context } from './CToast'; //component - CoreUI / CToastHeader

var CToastHeader = function CToastHeader(props) {
  var className = props.className,
      children = props.children,
      innerRef = props.innerRef,
      closeButton = props.closeButton,
      attributes = _objectWithoutPropertiesLoose(props, ["className", "children", "innerRef", "closeButton"]);

  var _useContext = useContext(Context),
      close = _useContext.close; //render


  var classes = classNames('toast-header', className);
  return /*#__PURE__*/React.createElement("div", _extends({
    className: classes
  }, attributes, {
    ref: innerRef
  }), children, closeButton && /*#__PURE__*/React.createElement(CButtonClose, {
    className: "ml-auto",
    onClick: close
  }));
};

CToastHeader.propTypes = process.env.NODE_ENV !== "production" ? {
  className: PropTypes.string,
  children: PropTypes.node,
  //
  innerRef: PropTypes.oneOfType([PropTypes.func, PropTypes.string, PropTypes.object]),
  closeButton: PropTypes.bool
} : {};
CToastHeader.defaultProps = {
  closeButton: true
};
export default CToastHeader;
