// @coreui/react/es/form/CInputGroupAddon.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames'; //component - CoreUI / CInputGroupAddon

var CInputGroupAddon = function CInputGroupAddon(props) {
  var children = props.children,
      className = props.className,
      innerRef = props.innerRef,
      prepend = props.prepend,
      attributes = _objectWithoutPropertiesLoose(props, ["children", "className", "innerRef", "prepend"]); //render


  var classes = classNames('input-group-' + (prepend ? 'prepend' : 'append'), className);
  return /*#__PURE__*/React.createElement("div", _extends({
    className: classes
  }, attributes, {
    ref: innerRef
  }), children);
};

CInputGroupAddon.propTypes = process.env.NODE_ENV !== "production" ? {
  children: PropTypes.node,
  className: PropTypes.string,
  //
  innerRef: PropTypes.oneOfType([PropTypes.object, PropTypes.func, PropTypes.string]),
  prepend: PropTypes.bool
} : {};
export var CInputGroupAppend = function CInputGroupAppend(props) {
  return /*#__PURE__*/React.createElement(CInputGroupAddon, _extends({}, props, {
    prepend: false
  }));
};
export var CInputGroupPrepend = function CInputGroupPrepend(props) {
  return /*#__PURE__*/React.createElement(CInputGroupAddon, _extends({}, props, {
    prepend: true
  }));
};
