// @coreui/react/lib/modal/CModal.js
"use strict";

exports.__esModule = true;
exports["default"] = exports.Context = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactTransitionGroup = require("react-transition-group");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Context = _react["default"].createContext({});

exports.Context = Context;

var getTransitionClass = function getTransitionClass(s) {
  return s === 'entering' ? 'd-block' : s === 'entered' ? 'show d-block' : s === 'exiting' ? 'd-block' : '';
}; //component - CoreUI / CModal


var CModal = function CModal(props) {
  var _classNames, _classNames2, _classNames3;

  var innerRef = props.innerRef,
      show = props.show,
      centered = props.centered,
      size = props.size,
      color = props.color,
      borderColor = props.borderColor,
      fade = props.fade,
      backdrop = props.backdrop,
      closeOnBackdrop = props.closeOnBackdrop,
      onOpened = props.onOpened,
      onClosed = props.onClosed,
      addContentClass = props.addContentClass,
      onClose = props.onClose,
      attributes = _objectWithoutPropertiesLoose(props, ["innerRef", "show", "centered", "size", "color", "borderColor", "fade", "backdrop", "closeOnBackdrop", "onOpened", "onClosed", "addContentClass", "onClose"]);

  var _useState = (0, _react.useState)(show),
      isOpen = _useState[0],
      setIsOpen = _useState[1];

  var modalClick = function modalClick(e) {
    return e.target.dataset.modal && closeOnBackdrop && close();
  };

  (0, _react.useEffect)(function () {
    setIsOpen(show);
  }, [show]);

  var close = function close() {
    onClose && onClose();
    setIsOpen(false);
  };

  var onEntered = function onEntered() {
    return onOpened && onOpened();
  };

  var onExited = function onExited() {
    return onClosed && onClosed();
  };

  var modalClasses = (0, _classnames["default"])('modal overflow-auto fade', (_classNames = {}, _classNames["modal-" + color] = color, _classNames));
  var dialogClasses = (0, _classnames["default"])('modal-dialog', (_classNames2 = {
    'modal-dialog-centered': centered
  }, _classNames2["modal-" + size] = size, _classNames2));
  var contentClasses = (0, _classnames["default"])('modal-content', (_classNames3 = {}, _classNames3["border-" + borderColor] = borderColor, _classNames3), addContentClass);
  var backdropClasses = (0, _classnames["default"])({
    'modal-backdrop': true,
    'fade': fade,
    'show': isOpen || fade
  });
  return /*#__PURE__*/_react["default"].createElement("div", {
    onClick: modalClick
  }, /*#__PURE__*/_react["default"].createElement(_reactTransitionGroup.Transition, {
    "in": Boolean(isOpen),
    onEntered: onEntered,
    onExited: onExited,
    timeout: fade ? 150 : 0
  }, function (status) {
    var transitionClass = getTransitionClass(status);
    var classes = (0, _classnames["default"])(modalClasses, transitionClass);
    return /*#__PURE__*/_react["default"].createElement("div", {
      tabIndex: "-1",
      role: "dialog",
      className: classes,
      "data-modal": true
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: dialogClasses,
      role: "document"
    }, /*#__PURE__*/_react["default"].createElement("div", _extends({}, attributes, {
      className: contentClasses,
      ref: innerRef
    }), /*#__PURE__*/_react["default"].createElement(Context.Provider, {
      value: {
        close: close
      }
    }, props.children))));
  }), backdrop && isOpen && /*#__PURE__*/_react["default"].createElement("div", {
    className: backdropClasses
  }));
};

CModal.propTypes = process.env.NODE_ENV !== "production" ? {
  children: _propTypes["default"].node,
  className: _propTypes["default"].string,
  innerRef: _propTypes["default"].oneOfType([_propTypes["default"].object, _propTypes["default"].func, _propTypes["default"].string]),
  show: _propTypes["default"].bool,
  centered: _propTypes["default"].bool,
  size: _propTypes["default"].oneOf(['', 'sm', 'lg', 'xl']),
  backdrop: _propTypes["default"].bool,
  color: _propTypes["default"].string,
  borderColor: _propTypes["default"].string,
  onOpened: _propTypes["default"].func,
  onClosed: _propTypes["default"].func,
  fade: _propTypes["default"].bool,
  closeOnBackdrop: _propTypes["default"].bool,
  onClose: _propTypes["default"].func,
  addContentClass: _propTypes["default"].string
} : {};
CModal.defaultProps = {
  backdrop: true,
  fade: true,
  closeOnBackdrop: true
};
var _default = CModal;
exports["default"] = _default;
