// @coreui/react/es/widgets/CWidgetProgress.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import CProgress from '../progress/CProgress'; //component - CoreUI / CWidgetProgress

var CWidgetProgress = function CWidgetProgress(props) {
  var children = props.children,
      className = props.className,
      header = props.header,
      text = props.text,
      footer = props.footer,
      color = props.color,
      value = props.value,
      inverse = props.inverse,
      attributes = _objectWithoutPropertiesLoose(props, ["children", "className", "header", "text", "footer", "color", "value", "inverse"]);

  var classes = classNames('card', inverse ? [color && "bg-" + color, 'text-white'] : '', className);
  return /*#__PURE__*/React.createElement("div", _extends({
    className: classes
  }, attributes), /*#__PURE__*/React.createElement("div", {
    className: "card-body"
  }, header && /*#__PURE__*/React.createElement("div", {
    className: "h4 m-0"
  }, header), text && /*#__PURE__*/React.createElement("div", null, text), children || /*#__PURE__*/React.createElement(CProgress, {
    color: !inverse ? color : '',
    value: value,
    className: "progress-xs my-3 mb-0 " + (inverse ? 'progress-white' : '')
  }), footer && /*#__PURE__*/React.createElement("small", {
    className: "text-muted"
  }, footer)));
};

CWidgetProgress.propTypes = process.env.NODE_ENV !== "production" ? {
  children: PropTypes.node,
  className: PropTypes.string,
  //
  innerRef: PropTypes.oneOfType([PropTypes.object, PropTypes.func, PropTypes.string]),
  header: PropTypes.string,
  text: PropTypes.string,
  footer: PropTypes.string,
  color: PropTypes.string,
  value: PropTypes.number,
  inverse: PropTypes.bool
} : {};
CWidgetProgress.defaultProps = {
  value: 25
};
export default CWidgetProgress;
