// @coreui/react/lib/button/CButtonToolbar.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

//component - CoreUI / CButtonToolbar
var CButtonToolbar = function CButtonToolbar(props) {
  var _classNames;

  var className = props.className,
      innerRef = props.innerRef,
      justify = props.justify,
      attributes = _objectWithoutPropertiesLoose(props, ["className", "innerRef", "justify"]); //render


  var classes = (0, _classnames["default"])(className, 'btn-toolbar', (_classNames = {}, _classNames["justify-content-" + justify] = justify, _classNames));
  return /*#__PURE__*/_react["default"].createElement("div", _extends({
    className: classes,
    role: "toolbar",
    "aria-label": "toolbar"
  }, attributes, {
    ref: innerRef
  }));
};

CButtonToolbar.propTypes = process.env.NODE_ENV !== "production" ? {
  className: _propTypes["default"].string,
  //
  innerRef: _propTypes["default"].oneOfType([_propTypes["default"].object, _propTypes["default"].func, _propTypes["default"].string]),
  role: _propTypes["default"].string,
  justify: _propTypes["default"].oneOf(['', 'start', 'end', 'between', 'center'])
} : {};
var _default = CButtonToolbar;
exports["default"] = _default;
module.exports = exports.default;
