// @coreui/react/lib/form/CFormFeedback.js
"use strict";

exports.__esModule = true;
exports.CInvalidFeedback = exports.CValidFeedback = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

//component - CoreUI / CFormFeedback
var CFormFeedback = function CFormFeedback(props) {
  var className = props.className,
      innerRef = props.innerRef,
      valid = props.valid,
      tooltip = props.tooltip,
      attributes = _objectWithoutPropertiesLoose(props, ["className", "innerRef", "valid", "tooltip"]); //render


  var validMode = tooltip ? 'tooltip' : 'feedback';
  var classes = (0, _classnames["default"])(valid ? "valid-" + validMode : "invalid-" + validMode, className);
  return /*#__PURE__*/_react["default"].createElement("div", _extends({
    className: classes
  }, attributes, {
    ref: innerRef
  }));
};

CFormFeedback.propTypes = process.env.NODE_ENV !== "production" ? {
  children: _propTypes["default"].node,
  className: _propTypes["default"].string,
  //
  innerRef: _propTypes["default"].oneOfType([_propTypes["default"].object, _propTypes["default"].func, _propTypes["default"].string]),
  valid: _propTypes["default"].bool,
  tooltip: _propTypes["default"].bool
} : {};

var CValidFeedback = function CValidFeedback(props) {
  return /*#__PURE__*/_react["default"].createElement(CFormFeedback, _extends({}, props, {
    valid: true
  }));
};

exports.CValidFeedback = CValidFeedback;

var CInvalidFeedback = function CInvalidFeedback(props) {
  return /*#__PURE__*/_react["default"].createElement(CFormFeedback, _extends({}, props, {
    valid: false
  }));
};

exports.CInvalidFeedback = CInvalidFeedback;
