// @coreui/react/lib/form/CLabel.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _helper = require("../utils/helper.js");

var _CCol = _interopRequireDefault(require("../grid/CCol"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

//component - CoreUI / CLabel
var CLabel = function CLabel(props) {
  var tag = props.tag,
      className = props.className,
      innerRef = props.innerRef,
      hidden = props.hidden,
      variant = props.variant,
      col = props.col,
      attributes = _objectWithoutPropertiesLoose(props, ["tag", "className", "innerRef", "hidden", "variant", "col"]); // render


  var classes = (0, _classnames["default"])(hidden && 'sr-only', variant === 'custom-checkbox' && 'custom-control-label', variant === 'checkbox' && 'form-check-label', variant === 'custom-file' && 'custom-file-label', col && 'col-form-label', col && typeof col === 'string' && "col-form-label-" + col, className);
  var Tag = col ? _CCol["default"] : tag;
  var addLabelTag = col && {
    tag: tag
  };
  return /*#__PURE__*/_react["default"].createElement(Tag, _extends({
    className: classes
  }, addLabelTag, attributes, {
    ref: innerRef
  }));
};

CLabel.propTypes = process.env.NODE_ENV !== "production" ? {
  tag: _helper.tagPropType,
  children: _propTypes["default"].node,
  className: _propTypes["default"].string,
  //
  innerRef: _propTypes["default"].oneOfType([_propTypes["default"].object, _propTypes["default"].func, _propTypes["default"].string]),
  hidden: _propTypes["default"].bool,
  variant: _propTypes["default"].oneOf(['custom-file', 'checkbox', 'custom-checkbox']),
  col: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].bool])
} : {};
CLabel.defaultProps = {
  tag: 'label'
};
var _default = CLabel;
exports["default"] = _default;
module.exports = exports.default;
