// @coreui/react/lib/card/CCard.js
"use strict";

exports.__esModule = true;
exports["default"] = exports.sharedPropTypes = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _helper = require("../utils/helper.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

//component - CoreUI / CCard
var CCard = function CCard(props) {
  var Tag = props.tag,
      className = props.className,
      innerRef = props.innerRef,
      color = props.color,
      textColor = props.textColor,
      borderColor = props.borderColor,
      align = props.align,
      accentColor = props.accentColor,
      attributes = _objectWithoutPropertiesLoose(props, ["tag", "className", "innerRef", "color", "textColor", "borderColor", "align", "accentColor"]); //render


  var classes = (0, _classnames["default"])(className, 'card', align ? "text-" + align : false, textColor ? "text-" + textColor : false, color ? "bg-" + color : false, borderColor ? "border-" + borderColor : false, accentColor ? "card-accent-" + accentColor : false);
  return /*#__PURE__*/_react["default"].createElement(Tag, _extends({
    className: classes
  }, attributes, {
    ref: innerRef
  }));
};

var sharedPropTypes = {
  align: _propTypes["default"].oneOf(['', 'left', 'center', 'right']),
  color: _propTypes["default"].string,
  borderColor: _propTypes["default"].string,
  textColor: _propTypes["default"].string
};
exports.sharedPropTypes = sharedPropTypes;
CCard.propTypes = process.env.NODE_ENV !== "production" ? _extends({
  tag: _helper.tagPropType,
  className: _propTypes["default"].string,
  //
  innerRef: _propTypes["default"].oneOfType([_propTypes["default"].object, _propTypes["default"].func, _propTypes["default"].string]),
  accentColor: _propTypes["default"].string
}, sharedPropTypes) : {};
CCard.defaultProps = {
  tag: 'div'
};
var _default = CCard;
exports["default"] = _default;
