// @coreui/react/lib/alert/CAlert.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _CFade = _interopRequireDefault(require("../fade/CFade"));

var _src = require("@coreui/utils/src");

var _helper = require("../utils/helper.js");

var _CButtonClose = _interopRequireDefault(require("../button/CButtonClose"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

//component - CoreUI / CAlert
var CAlert = function CAlert(props) {
  var _classNames;

  var children = props.children,
      className = props.className,
      onShowChange = props.onShowChange,
      closeButton = props.closeButton,
      color = props.color,
      fade = props.fade,
      show = props.show,
      attributes = _objectWithoutPropertiesLoose(props, ["children", "className", "onShowChange", "closeButton", "color", "fade", "show"]); //render


  var classes = (0, _classnames["default"])(className, 'alert', (_classNames = {}, _classNames["alert-" + color] = color, _classNames['alert-dismissible'] = closeButton, _classNames));
  var alertTransition = {
    baseClass: fade ? _CFade["default"].baseClass : '',
    timeout: fade ? _CFade["default"].timeout : 0,
    unmountOnExit: true
  };

  var _useState = (0, _react.useState)(show),
      isOpen = _useState[0],
      setIsOpen = _useState[1];

  (0, _react.useEffect)(function () {
    setIsOpen(show);
  }, [show]);
  var timeout = (0, _react.useRef)();
  (0, _react.useEffect)(function () {
    onShowChange && onShowChange(isOpen);
    clearTimeout(timeout.current);

    if (typeof isOpen === 'number' && isOpen > 0) {
      timeout.current = setTimeout(function () {
        setIsOpen(isOpen - 1);
      }, 1000);
    }

    return function () {
      return clearTimeout(timeout.current);
    };
  }, [isOpen]);
  var attrs = (0, _src.omitByKeys)(attributes, _helper.CFadeProps);
  return /*#__PURE__*/_react["default"].createElement(_CFade["default"], _extends({}, alertTransition, {
    className: classes,
    "in": Boolean(isOpen),
    role: "alert"
  }, attrs), children, closeButton && /*#__PURE__*/_react["default"].createElement(_CButtonClose["default"], {
    onClick: function onClick() {
      return setIsOpen(false);
    }
  }));
};

CAlert.propTypes = process.env.NODE_ENV !== "production" ? {
  children: _propTypes["default"].node,
  className: _propTypes["default"].string,
  //
  innerRef: _propTypes["default"].oneOfType([_propTypes["default"].object, _propTypes["default"].func, _propTypes["default"].string]),
  onShowChange: _propTypes["default"].func,
  closeButton: _propTypes["default"].bool,
  color: _propTypes["default"].string,
  fade: _propTypes["default"].bool,
  show: _propTypes["default"].oneOfType([_propTypes["default"].bool, _propTypes["default"].number])
} : {};
CAlert.defaultProps = {
  show: true,
  fade: true
};
var _default = CAlert;
exports["default"] = _default;
module.exports = exports.default;
