// @coreui/react/lib/tabs/CTabContent.js
"use strict";

exports.__esModule = true;
exports["default"] = exports.Context = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

//component - CoreUI / CTabContent
var Context = _react["default"].createContext();

exports.Context = Context;

var CTabContent = function CTabContent(props) {
  var className = props.className,
      innerRef = props.innerRef,
      fade = props.fade,
      attributes = _objectWithoutPropertiesLoose(props, ["className", "innerRef", "fade"]); // render


  var classes = (0, _classnames["default"])('tab-content', className);
  return /*#__PURE__*/_react["default"].createElement(Context.Provider, {
    value: fade
  }, /*#__PURE__*/_react["default"].createElement("div", _extends({
    className: classes
  }, attributes, {
    ref: innerRef
  })));
};

CTabContent.propTypes = process.env.NODE_ENV !== "production" ? {
  className: _propTypes["default"].string,
  //
  innerRef: _propTypes["default"].oneOfType([_propTypes["default"].object, _propTypes["default"].func, _propTypes["default"].string]),
  fade: _propTypes["default"].bool
} : {};
CTabContent.defaultProps = {
  fade: true
};
var _default = CTabContent;
exports["default"] = _default;
