// @coreui/react/lib/template/CSidebarClose.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _CButtonClose = _interopRequireDefault(require("../button/CButtonClose"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

//component - CoreUI / CSidebarClose
var CSidebarClose = function CSidebarClose(props) {
  return /*#__PURE__*/_react["default"].createElement(_CButtonClose["default"], _extends({}, props, {
    buttonClass: 'c-sidebar-close'
  }), /*#__PURE__*/_react["default"].createElement("svg", {
    className: "c-icon",
    width: "24",
    height: "24",
    viewBox: "0 0 24 24"
  }, /*#__PURE__*/_react["default"].createElement("title", null, "x"), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M20.030 5.030l-1.061-1.061-6.97 6.97-6.97-6.97-1.061 1.061 6.97 6.97-6.97 6.97 1.061 1.061 6.97-6.97 6.97 6.97 1.061-1.061-6.97-6.97 6.97-6.97z"
  })));
};

var _default = CSidebarClose;
exports["default"] = _default;
module.exports = exports.default;
