// @coreui/react/es/tooltip/CTooltip.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { useState, useEffect, useRef, useCallback } from 'react';
import PropTypes from 'prop-types';
import tippy from 'tippy.js';
import 'tippy.js/dist/tippy.css';
import { renderToString } from 'react-dom/server'; //component - CoreUI / CTooltip

var CTooltip = function CTooltip(props) {
  var children = props.children,
      content = props.content,
      interactive = props.interactive,
      placement = props.placement,
      trigger = props.trigger,
      advancedOptions = props.advancedOptions;
  var computedContent = useCallback(function () {
    return typeof content === 'string' ? content : renderToString(content);
  }, [content]);

  var config = _extends({
    allowHTML: true,
    content: computedContent,
    interactive: interactive,
    placement: placement,
    trigger: trigger
  }, advancedOptions);

  var key = useState(Math.random().toString(36).substr(2))[0];
  var instance = useRef();
  useEffect(function () {
    if (instance.current) {
      instance.current.setProps(config);
    }
  });
  useEffect(function () {
    var node = document.querySelector("[data-tooltip=\"" + key + "\"]");
    instance.current = tippy(node, config);
    return function () {
      return instance.current.destroy();
    };
  }, [key]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, React.cloneElement(children, {
    'data-tooltip': key
  }));
};

CTooltip.propTypes = process.env.NODE_ENV !== "production" ? {
  children: PropTypes.node,
  content: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  interactive: PropTypes.bool,
  placement: PropTypes.oneOf(['', 'top-end', 'top', 'top-start', 'bottom-end', 'bottom', 'bottom-start', 'right-start', 'right', 'right-end', 'left-start', 'left', 'left-end']),
  trigger: PropTypes.string,
  advancedOptions: PropTypes.object
} : {};
CTooltip.defaultProps = {
  content: '',
  interactive: false,
  placement: 'top',
  trigger: 'mouseenter focus',
  advancedOptions: {}
};
export default CTooltip;
