// @coreui/react/lib/index.js
"use strict";

exports.__esModule = true;
exports.CToggler = exports.CToaster = exports.CToastBody = exports.CToastHeader = exports.CToast = exports.CSidebarClose = exports.CSidebarNav = exports.CSidebarMinimizer = exports.CSidebarHeader = exports.CSidebarForm = exports.CSidebarFooter = exports.CSidebarNavDropdown = exports.CSidebarNavItem = exports.CSidebarNavTitle = exports.CSidebarNavDivider = exports.CSidebarBrand = exports.CSidebar = exports.CSubheader = exports.CHeaderBrand = exports.CHeaderNavLink = exports.CHeaderNavItem = exports.CHeaderNav = exports.CHeader = exports.CFooter = exports.CTabContent = exports.CTabPane = exports.CTabs = exports.CDataTable = exports.CSwitch = exports.CSpinner = exports.CProgressBar = exports.CProgress = exports.CPagination = exports.CNavbarText = exports.CNavbarNav = exports.CNavbar = exports.CNavbarBrand = exports.CNavLink = exports.CNavItem = exports.CNav = exports.CModalTitle = exports.CModalFooter = exports.CModalHeader = exports.CModalBody = exports.CModal = exports.CMediaBody = exports.CMedia = exports.CListGroupItem = exports.CListGroup = exports.CLink = exports.CJumbotron = exports.CImg = exports.CContainer = exports.CCol = exports.CRow = exports.CInputGroupText = exports.CInputGroup = exports.CFormText = exports.CFormGroup = exports.CLabel = exports.CForm = exports.CFade = exports.CEmbedItem = exports.CEmbed = exports.CElementCover = exports.CDropdownToggle = exports.CDropdownMenu = exports.CDropdownItem = exports.CDropdownHeader = exports.CDropdownDivider = exports.CDropdown = exports.CCreateElement = exports.CCollapse = exports.CCarouselInner = exports.CCarouselCaption = exports.CCarouselIndicators = exports.CCarouselControl = exports.CCarouselItem = exports.CCarousel = exports.CCardText = exports.CCardSubtitle = exports.CCardTitle = exports.CCardLink = exports.CCardImgOverlay = exports.CCardImg = exports.CCardGroup = exports.CCardFooter = exports.CCardHeader = exports.CCardBody = exports.CCard = exports.CCallout = exports.CBreadcrumbItem = exports.CBreadcrumbRouter = exports.CBreadcrumb = exports.CButtonToolbar = exports.CButtonGroup = exports.CButtonClose = exports.CButton = exports.CBadge = exports.CAlert = void 0;
exports.CListGroupItemHeading = exports.CListGroupItemText = exports.CInputGroupAppend = exports.CInputGroupPrepend = exports.CSelect = exports.CInputRadio = exports.CInputCheckbox = exports.CInputFile = exports.CInput = exports.CTextarea = exports.CInvalidFeedback = exports.CValidFeedback = exports.CWidgetSimple = exports.CWidgetDropdown = exports.CWidgetProgressIcon = exports.CWidgetBrand = exports.CWidgetIcon = exports.CWidgetProgress = exports.CPopover = exports.CTooltip = void 0;

var _CAlert = _interopRequireDefault(require("./alert/CAlert"));

exports.CAlert = _CAlert["default"];

var _CBadge = _interopRequireDefault(require("./badge/CBadge"));

exports.CBadge = _CBadge["default"];

var _CButton = _interopRequireDefault(require("./button/CButton"));

exports.CButton = _CButton["default"];

var _CButtonClose = _interopRequireDefault(require("./button/CButtonClose"));

exports.CButtonClose = _CButtonClose["default"];

var _CButtonGroup = _interopRequireDefault(require("./button/CButtonGroup"));

exports.CButtonGroup = _CButtonGroup["default"];

var _CButtonToolbar = _interopRequireDefault(require("./button/CButtonToolbar"));

exports.CButtonToolbar = _CButtonToolbar["default"];

var _CBreadcrumb = _interopRequireDefault(require("./breadcrumb/CBreadcrumb"));

exports.CBreadcrumb = _CBreadcrumb["default"];

var _CBreadcrumbRouter = _interopRequireDefault(require("./breadcrumb/CBreadcrumbRouter"));

exports.CBreadcrumbRouter = _CBreadcrumbRouter["default"];

var _CBreadcrumbItem = _interopRequireDefault(require("./breadcrumb/CBreadcrumbItem"));

exports.CBreadcrumbItem = _CBreadcrumbItem["default"];

var _CCallout = _interopRequireDefault(require("./callout/CCallout"));

exports.CCallout = _CCallout["default"];

var _CCard = _interopRequireDefault(require("./card/CCard"));

exports.CCard = _CCard["default"];

var _CCardBody = _interopRequireDefault(require("./card/CCardBody"));

exports.CCardBody = _CCardBody["default"];

var _CCardHeader = _interopRequireDefault(require("./card/CCardHeader"));

exports.CCardHeader = _CCardHeader["default"];

var _CCardFooter = _interopRequireDefault(require("./card/CCardFooter"));

exports.CCardFooter = _CCardFooter["default"];

var _CCardGroup = _interopRequireDefault(require("./card/CCardGroup"));

exports.CCardGroup = _CCardGroup["default"];

var _CCardImg = _interopRequireDefault(require("./card/CCardImg"));

exports.CCardImg = _CCardImg["default"];

var _CCardImgOverlay = _interopRequireDefault(require("./card/CCardImgOverlay"));

exports.CCardImgOverlay = _CCardImgOverlay["default"];

var _CCardLink = _interopRequireDefault(require("./card/CCardLink"));

exports.CCardLink = _CCardLink["default"];

var _CCardTitle = _interopRequireDefault(require("./card/CCardTitle"));

exports.CCardTitle = _CCardTitle["default"];

var _CCardSubtitle = _interopRequireDefault(require("./card/CCardSubtitle"));

exports.CCardSubtitle = _CCardSubtitle["default"];

var _CCardText = _interopRequireDefault(require("./card/CCardText"));

exports.CCardText = _CCardText["default"];

var _CCarousel = _interopRequireDefault(require("./carousel/CCarousel"));

exports.CCarousel = _CCarousel["default"];

var _CCarouselItem = _interopRequireDefault(require("./carousel/CCarouselItem"));

exports.CCarouselItem = _CCarouselItem["default"];

var _CCarouselControl = _interopRequireDefault(require("./carousel/CCarouselControl"));

exports.CCarouselControl = _CCarouselControl["default"];

var _CCarouselIndicators = _interopRequireDefault(require("./carousel/CCarouselIndicators"));

exports.CCarouselIndicators = _CCarouselIndicators["default"];

var _CCarouselCaption = _interopRequireDefault(require("./carousel/CCarouselCaption"));

exports.CCarouselCaption = _CCarouselCaption["default"];

var _CCarouselInner = _interopRequireDefault(require("./carousel/CCarouselInner"));

exports.CCarouselInner = _CCarouselInner["default"];

var _CCollapse = _interopRequireDefault(require("./collapse/CCollapse"));

exports.CCollapse = _CCollapse["default"];

var _CCreateElement = _interopRequireDefault(require("./create-element/CCreateElement"));

exports.CCreateElement = _CCreateElement["default"];

var _CDropdown = _interopRequireDefault(require("./dropdown/CDropdown"));

exports.CDropdown = _CDropdown["default"];

var _CDropdownDivider = _interopRequireDefault(require("./dropdown/CDropdownDivider"));

exports.CDropdownDivider = _CDropdownDivider["default"];

var _CDropdownHeader = _interopRequireDefault(require("./dropdown/CDropdownHeader"));

exports.CDropdownHeader = _CDropdownHeader["default"];

var _CDropdownItem = _interopRequireDefault(require("./dropdown/CDropdownItem"));

exports.CDropdownItem = _CDropdownItem["default"];

var _CDropdownMenu = _interopRequireDefault(require("./dropdown/CDropdownMenu"));

exports.CDropdownMenu = _CDropdownMenu["default"];

var _CDropdownToggle = _interopRequireDefault(require("./dropdown/CDropdownToggle"));

exports.CDropdownToggle = _CDropdownToggle["default"];

var _CElementCover = _interopRequireDefault(require("./element-cover/CElementCover"));

exports.CElementCover = _CElementCover["default"];

var _CEmbed = _interopRequireDefault(require("./embed/CEmbed"));

exports.CEmbed = _CEmbed["default"];

var _CEmbedItem = _interopRequireDefault(require("./embed/CEmbedItem"));

exports.CEmbedItem = _CEmbedItem["default"];

var _CFade = _interopRequireDefault(require("./fade/CFade"));

exports.CFade = _CFade["default"];

var _CForm = _interopRequireDefault(require("./form/CForm"));

exports.CForm = _CForm["default"];

var _CFormFeedback = require("./form/CFormFeedback");

exports.CValidFeedback = _CFormFeedback.CValidFeedback;
exports.CInvalidFeedback = _CFormFeedback.CInvalidFeedback;

var _CLabel = _interopRequireDefault(require("./form/CLabel"));

exports.CLabel = _CLabel["default"];

var _CFormGroup = _interopRequireDefault(require("./form/CFormGroup"));

exports.CFormGroup = _CFormGroup["default"];

var _CFormText = _interopRequireDefault(require("./form/CFormText"));

exports.CFormText = _CFormText["default"];

var _CInput = require("./form/CInput");

exports.CTextarea = _CInput.CTextarea;
exports.CInput = _CInput.CInput;
exports.CInputFile = _CInput.CInputFile;
exports.CInputCheckbox = _CInput.CInputCheckbox;
exports.CInputRadio = _CInput.CInputRadio;
exports.CSelect = _CInput.CSelect;

var _CInputGroup = _interopRequireDefault(require("./form/CInputGroup"));

exports.CInputGroup = _CInputGroup["default"];

var _CInputGroupAddon = require("./form/CInputGroupAddon");

exports.CInputGroupPrepend = _CInputGroupAddon.CInputGroupPrepend;
exports.CInputGroupAppend = _CInputGroupAddon.CInputGroupAppend;

var _CInputGroupText = _interopRequireDefault(require("./form/CInputGroupText"));

exports.CInputGroupText = _CInputGroupText["default"];

var _CRow = _interopRequireDefault(require("./grid/CRow"));

exports.CRow = _CRow["default"];

var _CCol = _interopRequireDefault(require("./grid/CCol"));

exports.CCol = _CCol["default"];

var _CContainer = _interopRequireDefault(require("./grid/CContainer"));

exports.CContainer = _CContainer["default"];

var _CImg = _interopRequireDefault(require("./image/CImg"));

exports.CImg = _CImg["default"];

var _CJumbotron = _interopRequireDefault(require("./jumbotron/CJumbotron"));

exports.CJumbotron = _CJumbotron["default"];

var _CLink = _interopRequireDefault(require("./link/CLink"));

exports.CLink = _CLink["default"];

var _CListGroup = _interopRequireDefault(require("./list-group/CListGroup"));

exports.CListGroup = _CListGroup["default"];

var _CListGroupItem = _interopRequireDefault(require("./list-group/CListGroupItem"));

exports.CListGroupItem = _CListGroupItem["default"];

var _react = _interopRequireDefault(require("react"));

var _CMedia = _interopRequireDefault(require("./media/CMedia"));

exports.CMedia = _CMedia["default"];

var _CMediaBody = _interopRequireDefault(require("./media/CMediaBody"));

exports.CMediaBody = _CMediaBody["default"];

var _CModal = _interopRequireDefault(require("./modal/CModal"));

exports.CModal = _CModal["default"];

var _CModalBody = _interopRequireDefault(require("./modal/CModalBody"));

exports.CModalBody = _CModalBody["default"];

var _CModalHeader = _interopRequireDefault(require("./modal/CModalHeader"));

exports.CModalHeader = _CModalHeader["default"];

var _CModalFooter = _interopRequireDefault(require("./modal/CModalFooter"));

exports.CModalFooter = _CModalFooter["default"];

var _CModalTitle = _interopRequireDefault(require("./modal/CModalTitle"));

exports.CModalTitle = _CModalTitle["default"];

var _CNav = _interopRequireDefault(require("./nav/CNav"));

exports.CNav = _CNav["default"];

var _CNavItem = _interopRequireDefault(require("./nav/CNavItem"));

exports.CNavItem = _CNavItem["default"];

var _CNavLink = _interopRequireDefault(require("./nav/CNavLink"));

exports.CNavLink = _CNavLink["default"];

var _CNavbarBrand = _interopRequireDefault(require("./navbar/CNavbarBrand"));

exports.CNavbarBrand = _CNavbarBrand["default"];

var _CNavbar = _interopRequireDefault(require("./navbar/CNavbar"));

exports.CNavbar = _CNavbar["default"];

var _CNavbarNav = _interopRequireDefault(require("./navbar/CNavbarNav"));

exports.CNavbarNav = _CNavbarNav["default"];

var _CNavbarText = _interopRequireDefault(require("./navbar/CNavbarText"));

exports.CNavbarText = _CNavbarText["default"];

var _CPagination = _interopRequireDefault(require("./pagination/CPagination"));

exports.CPagination = _CPagination["default"];

var _CProgress = _interopRequireDefault(require("./progress/CProgress"));

exports.CProgress = _CProgress["default"];

var _CProgressBar = _interopRequireDefault(require("./progress/CProgressBar"));

exports.CProgressBar = _CProgressBar["default"];

var _CSpinner = _interopRequireDefault(require("./spinner/CSpinner"));

exports.CSpinner = _CSpinner["default"];

var _CSwitch = _interopRequireDefault(require("./switch/CSwitch"));

exports.CSwitch = _CSwitch["default"];

var _CDataTable = _interopRequireDefault(require("./table/CDataTable"));

exports.CDataTable = _CDataTable["default"];

var _CTabs = _interopRequireDefault(require("./tabs/CTabs"));

exports.CTabs = _CTabs["default"];

var _CTabPane = _interopRequireDefault(require("./tabs/CTabPane"));

exports.CTabPane = _CTabPane["default"];

var _CTabContent = _interopRequireDefault(require("./tabs/CTabContent"));

exports.CTabContent = _CTabContent["default"];

var _CFooter = _interopRequireDefault(require("./template/CFooter"));

exports.CFooter = _CFooter["default"];

var _CHeader = _interopRequireDefault(require("./template/CHeader"));

exports.CHeader = _CHeader["default"];

var _CHeaderNav = _interopRequireDefault(require("./template/CHeaderNav"));

exports.CHeaderNav = _CHeaderNav["default"];

var _CHeaderNavItem = _interopRequireDefault(require("./template/CHeaderNavItem"));

exports.CHeaderNavItem = _CHeaderNavItem["default"];

var _CHeaderNavLink = _interopRequireDefault(require("./template/CHeaderNavLink"));

exports.CHeaderNavLink = _CHeaderNavLink["default"];

var _CHeaderBrand = _interopRequireDefault(require("./template/CHeaderBrand"));

exports.CHeaderBrand = _CHeaderBrand["default"];

var _CSubheader = _interopRequireDefault(require("./template/CSubheader"));

exports.CSubheader = _CSubheader["default"];

var _CSidebar = _interopRequireDefault(require("./template/CSidebar"));

exports.CSidebar = _CSidebar["default"];

var _CSidebarBrand = _interopRequireDefault(require("./template/CSidebarBrand"));

exports.CSidebarBrand = _CSidebarBrand["default"];

var _CSidebarNavDivider = _interopRequireDefault(require("./template/CSidebarNavDivider"));

exports.CSidebarNavDivider = _CSidebarNavDivider["default"];

var _CSidebarNavTitle = _interopRequireDefault(require("./template/CSidebarNavTitle"));

exports.CSidebarNavTitle = _CSidebarNavTitle["default"];

var _CSidebarNavItem = _interopRequireDefault(require("./template/CSidebarNavItem"));

exports.CSidebarNavItem = _CSidebarNavItem["default"];

var _CSidebarNavDropdown = _interopRequireDefault(require("./template/CSidebarNavDropdown"));

exports.CSidebarNavDropdown = _CSidebarNavDropdown["default"];

var _CSidebarFooter = _interopRequireDefault(require("./template/CSidebarFooter"));

exports.CSidebarFooter = _CSidebarFooter["default"];

var _CSidebarForm = _interopRequireDefault(require("./template/CSidebarForm"));

exports.CSidebarForm = _CSidebarForm["default"];

var _CSidebarHeader = _interopRequireDefault(require("./template/CSidebarHeader"));

exports.CSidebarHeader = _CSidebarHeader["default"];

var _CSidebarMinimizer = _interopRequireDefault(require("./template/CSidebarMinimizer"));

exports.CSidebarMinimizer = _CSidebarMinimizer["default"];

var _CSidebarNav = _interopRequireDefault(require("./template/CSidebarNav"));

exports.CSidebarNav = _CSidebarNav["default"];

var _CSidebarClose = _interopRequireDefault(require("./template/CSidebarClose"));

exports.CSidebarClose = _CSidebarClose["default"];

var _CToast = _interopRequireDefault(require("./toast/CToast"));

exports.CToast = _CToast["default"];

var _CToastHeader = _interopRequireDefault(require("./toast/CToastHeader"));

exports.CToastHeader = _CToastHeader["default"];

var _CToastBody = _interopRequireDefault(require("./toast/CToastBody"));

exports.CToastBody = _CToastBody["default"];

var _CToaster = _interopRequireDefault(require("./toast/CToaster"));

exports.CToaster = _CToaster["default"];

var _CToggler = _interopRequireDefault(require("./toggler/CToggler"));

exports.CToggler = _CToggler["default"];

var _CTooltip = _interopRequireDefault(require("./tooltip/CTooltip"));

exports.CTooltip = _CTooltip["default"];

var _CPopover = _interopRequireDefault(require("./tooltip/CPopover"));

exports.CPopover = _CPopover["default"];

var _CWidgetProgress = _interopRequireDefault(require("./widgets/CWidgetProgress"));

exports.CWidgetProgress = _CWidgetProgress["default"];

var _CWidgetIcon = _interopRequireDefault(require("./widgets/CWidgetIcon"));

exports.CWidgetIcon = _CWidgetIcon["default"];

var _CWidgetBrand = _interopRequireDefault(require("./widgets/CWidgetBrand"));

exports.CWidgetBrand = _CWidgetBrand["default"];

var _CWidgetProgressIcon = _interopRequireDefault(require("./widgets/CWidgetProgressIcon"));

exports.CWidgetProgressIcon = _CWidgetProgressIcon["default"];

var _CWidgetDropdown = _interopRequireDefault(require("./widgets/CWidgetDropdown"));

exports.CWidgetDropdown = _CWidgetDropdown["default"];

var _CWidgetSimple = _interopRequireDefault(require("./widgets/CWidgetSimple"));

exports.CWidgetSimple = _CWidgetSimple["default"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var wrapper = function wrapper(props) {
  console.warn('CListGroupItemText component is deprecated, please use div tag instead');
  return /*#__PURE__*/_react["default"].createElement("div", props);
};

exports.CListGroupItemText = wrapper;

var heading = function heading(props) {
  console.warn('CListGroupItemHeading component is deprecated, please use h5 tag instead');
  return /*#__PURE__*/_react["default"].createElement("h5", props);
};

exports.CListGroupItemHeading = heading;
