// @coreui/react/lib/widgets/CWidgetDropdown.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

//component - CoreUI / CWidgetDropdown
var CWidgetDropdown = function CWidgetDropdown(props) {
  var children = props.children,
      className = props.className,
      header = props.header,
      color = props.color,
      footerSlot = props.footerSlot,
      text = props.text,
      attributes = _objectWithoutPropertiesLoose(props, ["children", "className", "header", "color", "footerSlot", "text"]);

  var classes = (0, _classnames["default"])('card text-white', color && "bg-" + color, className);
  return /*#__PURE__*/_react["default"].createElement("div", _extends({
    className: classes
  }, attributes), /*#__PURE__*/_react["default"].createElement("div", {
    className: "card-body pb-0 d-flex justify-content-between"
  }, /*#__PURE__*/_react["default"].createElement("div", null, header && /*#__PURE__*/_react["default"].createElement("div", {
    className: "text-value-lg"
  }, header), text && /*#__PURE__*/_react["default"].createElement("div", null, text)), children), footerSlot);
};

CWidgetDropdown.propTypes = process.env.NODE_ENV !== "production" ? {
  children: _propTypes["default"].node,
  className: _propTypes["default"].string,
  //
  innerRef: _propTypes["default"].oneOfType([_propTypes["default"].object, _propTypes["default"].func, _propTypes["default"].string]),
  header: _propTypes["default"].string,
  color: _propTypes["default"].string,
  footerSlot: _propTypes["default"].node,
  text: _propTypes["default"].string
} : {};
var _default = CWidgetDropdown;
exports["default"] = _default;
module.exports = exports.default;
