// @coreui/react/lib/widgets/CWidgetBrand.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

//component - CoreUI / CWidgetBrand
var CWidgetBrand = function CWidgetBrand(props) {
  var children = props.children,
      className = props.className,
      color = props.color,
      rightHeader = props.rightHeader,
      rightFooter = props.rightFooter,
      leftHeader = props.leftHeader,
      leftFooter = props.leftFooter,
      addHeaderClasses = props.addHeaderClasses,
      bodySlot = props.bodySlot,
      attributes = _objectWithoutPropertiesLoose(props, ["children", "className", "color", "rightHeader", "rightFooter", "leftHeader", "leftFooter", "addHeaderClasses", "bodySlot"]); // render


  var headerClasses = (0, _classnames["default"])('card-header content-center text-white p-0', color && "bg-" + color, addHeaderClasses);
  return /*#__PURE__*/_react["default"].createElement("div", _extends({
    className: "card " + className
  }, attributes), /*#__PURE__*/_react["default"].createElement("div", {
    className: headerClasses
  }, children), bodySlot || /*#__PURE__*/_react["default"].createElement("div", {
    className: "card-body row text-center"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "col"
  }, rightHeader && /*#__PURE__*/_react["default"].createElement("div", {
    className: "text-value-lg"
  }, rightHeader), rightFooter && /*#__PURE__*/_react["default"].createElement("div", {
    className: "text-uppercase text-muted small"
  }, rightFooter)), /*#__PURE__*/_react["default"].createElement("div", {
    className: "c-vr"
  }), /*#__PURE__*/_react["default"].createElement("div", {
    className: "col"
  }, leftHeader && /*#__PURE__*/_react["default"].createElement("div", {
    className: "text-value-lg"
  }, leftHeader), leftFooter && /*#__PURE__*/_react["default"].createElement("div", {
    className: "text-uppercase text-muted small"
  }, leftFooter))));
};

CWidgetBrand.propTypes = process.env.NODE_ENV !== "production" ? {
  children: _propTypes["default"].node,
  className: _propTypes["default"].string,
  //
  innerRef: _propTypes["default"].oneOfType([_propTypes["default"].object, _propTypes["default"].func, _propTypes["default"].string]),
  color: _propTypes["default"].string,
  rightHeader: _propTypes["default"].string,
  rightFooter: _propTypes["default"].string,
  leftHeader: _propTypes["default"].string,
  leftFooter: _propTypes["default"].string,
  addHeaderClasses: _propTypes["default"].oneOfType([String, Array, Object]),
  bodySlot: _propTypes["default"].node
} : {};
var _default = CWidgetBrand;
exports["default"] = _default;
module.exports = exports.default;
