// @coreui/react/lib/breadcrumb/CBreadcrumbRouter.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _index = require("../index");

var _reactRouterDom = require("react-router-dom");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

//component - CoreUI / CBreadcrumbRouter
var getPaths = function getPaths(pathname) {
  var paths = ['/'];
  if (pathname === '/') return paths;
  pathname.split('/').reduce(function (prev, curr) {
    var currPath = prev + "/" + curr;
    paths.push(currPath);
    return currPath;
  });
  return paths;
};

var CBreadcrumbRouteItem = function CBreadcrumbRouteItem(_ref, currPath) {
  var name = _ref.name,
      path = _ref.path;

  if (path === currPath) {
    return /*#__PURE__*/_react["default"].createElement(_index.CBreadcrumbItem, {
      key: path,
      active: true
    }, name);
  } else {
    return /*#__PURE__*/_react["default"].createElement(_index.CBreadcrumbItem, {
      key: path
    }, /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Link, {
      to: path
    }, name));
  }
};

var CBreadcrumbRouter = function CBreadcrumbRouter(props) {
  var className = props.className,
      innerRef = props.innerRef,
      routes = props.routes,
      attributes = _objectWithoutPropertiesLoose(props, ["className", "innerRef", "routes"]);

  var items = null;

  if (routes) {
    var currPath = (0, _reactRouterDom.useLocation)().pathname;
    var paths = getPaths(currPath);
    var currRoutes = paths.map(function (path) {
      return routes.find(function (route) {
        return (0, _reactRouterDom.matchPath)(path, {
          path: route.path,
          exact: route.exact
        });
      });
    }).filter(function (route) {
      return route;
    });
    items = currRoutes.map(function (route) {
      return CBreadcrumbRouteItem(route, currPath);
    });
  } //render


  var classes = (0, _classnames["default"])(className);
  return /*#__PURE__*/_react["default"].createElement(_index.CBreadcrumb, _extends({
    className: classes
  }, attributes, {
    ref: innerRef
  }), items);
};

CBreadcrumbRouter.propTypes = process.env.NODE_ENV !== "production" ? {
  className: _propTypes["default"].string,
  innerRef: _propTypes["default"].oneOfType([_propTypes["default"].object, _propTypes["default"].func, _propTypes["default"].string]),
  routes: _propTypes["default"].array
} : {};
var _default = CBreadcrumbRouter;
exports["default"] = _default;
module.exports = exports.default;
